### PR TITLE
Enable testGetsWorkingGroupLens

### PR DESCRIPTION
**Pull request by mkondratek**
Related to https://linear.app/sourcegraph/issue/CODY-1909/ci-tests-are-flaky#comment-da664388.

Interesting. I ma sure the test failed for me locally and in CI. Possibly it is just flaky 🫤 

I will look into it 🕵️ 

## Test plan
1. Green CI

### CODE REVIEW

**Comment by mkondratek** [comment]
```
Lens group should have 8 widgets
Expected :8
Actual   :11
```
```
widgets: [
  LensIcon(icon=UrlResolver{ownerClass=com.sourcegraph.cody.Icons, classLoader=com.intellij.util.lang.PathClassLoader@5ebec15, overriddenPath='/icons/codyLogoMonochromatic.svg', url=file:/unresolved, useCacheOnLoad=true}), 
  LensLabel(text= ), 
  LensAction(text=Accept), 
  LensLabel(text= ∣ ), 
  LensAction(text=Undo), 
  LensHotkey(text=⌃⇧⌫), 
  LensLabel(text= ∣ ), 
  LensAction(text=Edit & Retry), 
  LensHotkey(text=⌃⇧⏎), 
  LensLabel(text= ∣ ), 
  LensAction(text=Show Diff)
]
```

**Comment by mkondratek** [comment]
The problem is that sometimes at the moment we check the lens group the task is already in `Applied` state and it is showing the accept group.

File: src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt [hunk 1]
```diff
@@ -37,28 +40,45 @@ class DocumentCodeTest : CodyIntegrationTextFixture() {
...
 
-  fun skip_testGetsWorkingGroupLens() {
+  fun testGetsWorkingGroupLens() {
+    var lenses: LensWidgetGroup? = null
+    var widgets: MutableList<LensWidget>? = null
+    val showWorkingGroupSessionStateListener =
+        object : FixupService.ActiveFixupSessionStateListener {
+          override fun fixupSessionStateChanged(isInProgress: Boolean) {
+            if (isInProgress) {
+              TestCase.assertNull("Value already set", lenses)
```
Comment (pkukielka) [inline src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt]: Assert messages could be more descriptive so if it fails it is immediately obvious why.

File: src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt [hunk 2]
```diff
@@ -37,28 +40,45 @@ class DocumentCodeTest : CodyIntegrationTextFixture() {
...
+          }
+        }
+    FixupService.getInstance(project).addListener(showWorkingGroupSessionStateListener)
     runAndWaitForNotifications(DocumentCodeAction.ID, TOPIC_DISPLAY_WORKING_GROUP)
     assertInlayIsShown()
 
     // Lens group should match the expected structure.
-    val lenses = activeSession().lensGroup
     assertNotNull("Lens group should be displayed", lenses)
+    val theWidgets = widgets!!
```
Comment (pkukielka) [inline src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt]: Using `the` in variable names seems a bit unusual to me :) But it does not matter too much

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**